lib/Doctrine/DBAL/Types/Type.php

```diff
@@ -148,6 +142,26 @@ public function getDefaultLength(AbstractPlatform $platform)
      */
     abstract public function getName();
 
+    private static function getTypeRegistry() : TypeRegistry
+    {
+        if (self::$typeRegistry === null) {
+            return self::$typeRegistry = self::createTypeRegistry();
+        }
+
+        return self::$typeRegistry;
+    }
+
+    private static function createTypeRegistry() : TypeRegistry
+    {
+        $registry = new TypeRegistry();
+
+        foreach (self::BUILTIN_TYPES_MAP as $name => $class) {
+            $registry->addType($name, $class);
+        }
+
+        return $registry;
+    }
```

Comment:
Why make this a separate method? It's only used once I think.

Reply:
Readability.